tests/apps/demultiplex/test_sample_models.py

```diff
@@ -0,0 +1,398 @@
+)
+def test_get_index2_override_cycles(
+    lims_index: str, index2_cycles: int, reverse_cycle: bool, expected_parsed_cycles: str
+):
+    """Test that the returned index 2 cycles is the expected for different index configurations."""
+    # GIVEN a FlowCellSampleBCLConvert with separated indexes
+    sample = FlowCellSampleBCLConvert(lane=1, index=lims_index, sample_id="ACC123")
+    is_run_single_index: bool = not bool(index2_cycles)
+    sample.separate_indexes(is_run_single_index=is_run_single_index)
+
+    # WHEN getting the index2 override cycles
+    index2_cycles: str = sample._get_index2_override_cycles(
+        len_index2_cycles=index2_cycles, reverse_cycle=reverse_cycle
+    )
+
+    # THEN the index2 override cycles value is the expected one
+    assert index2_cycles == expected_parsed_cycles
+
+
+@pytest.mark.parametrize(
+    "run_parameters_fixture",
+    [
+        "hiseq_x_single_index_run_parameters",
+        "hiseq_x_dual_index_run_parameters",
+        "hiseq_2500_dual_index_run_parameters",
+        "hiseq_2500_custom_index_run_parameters",
+        "novaseq_6000_run_parameters_pre_1_5_kits",
+        "novaseq_6000_run_parameters_post_1_5_kits",
+        "novaseq_x_run_parameters",
+    ],
+)
+def test_update_override_cycles(
+    bcl_convert_flow_cell_sample: FlowCellSampleBCLConvert,
+    run_parameters_fixture: str,
+    request: pytest.FixtureRequest,
+):
+    """Test that updating a sample's override cycles works for different run parameters objects."""
+    # GIVEN a run parameters object
+    run_parameters: RunParameters = request.getfixturevalue(run_parameters_fixture)
+
+    # GIVEN a FlowCellSampleBCLConvert without override cycles
+    assert bcl_convert_flow_cell_sample.override_cycles == EMPTY_STRING
+
+    # WHEN updating the override cycles
+    bcl_convert_flow_cell_sample.update_override_cycles(run_parameters=run_parameters)
+
+    # THEN the override cycles are updated with the expected value
+    cycles: list[str] = bcl_convert_flow_cell_sample.override_cycles.split(";")
+    assert len(cycles) >= 3
+    assert cycles[0] == f"Y{run_parameters.get_read_1_cycles()}"
+    assert cycles[-1] == f"Y{run_parameters.get_read_2_cycles()}"
+    assert "I" in cycles[1]
+
+
+@pytest.mark.parametrize(
+    "sample_list_fixture, expected_barcode_mismatch",
+    [("bcl_convert_samples_similar_index1", 0), ("bcl_convert_samples_similar_index2", 1)],
+    ids=["barcode1_0", "barcode1_1"],
+)
+def test_update_barcode_mismatches_1(
+    sample_list_fixture: str, expected_barcode_mismatch: int, request: pytest.FixtureRequest
+):
+    """Test that index 1 barcode mismatch values are as expected for different sets of samples."""
+    # GIVEN a list of FlowCellSampleBCLConvert
+    sample_list: list[FlowCellSampleBCLConvert] = request.getfixturevalue(sample_list_fixture)
+
+    # GIVEN a FlowCellSampleBCLConvert
+    sample_to_update: FlowCellSampleBCLConvert = sample_list[0]
+
+    # WHEN updating the barcode mismatches 1
```

Comment:
Maybe be a bit more verbose here. Mismatches_1 is the variable name. In comments "mismatches for index 1" is a lot clearer. (Same for several other comments nearby)
```suggestion
    # WHEN updating the barcode mismatches 1
```

Reply:
updated all barcode mismatch comments